libs/lensfun/database.cpp

```diff
@@ -94,11 +94,11 @@ lfError lfDatabase::Load ()
 {
   lfError err = LF_NO_ERROR;
 
-    const int timestamp_system =
+    const auto timestamp_system =
```

Comment:
looks good. I like beeing explicit about the data types but this will work fine :)

Reply:
I assume the change was tested on a system with date set past 2038?

Reply:
> I assume the change was tested on a system with date set past 2038?

Yes, i did some basic testing.
The system date doesn't matter, it's about the value in in the `timstamp.txt` file in the database. With the change a timestamp after 2038 is no longer interpreted as a negative number.

Reply:
Unfortunately it does not fix the problem for MSVC, because `sizeof(long int)` is 4 bytes for this compiler even for 64bit systems and `ReadTimestamp` returns `long int`